src/Forms/Form.php

```diff
@@ -1405,9 +1413,10 @@ public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null)
             $submitted = true;
         }
 
-        // dont include fields without data
-        $dataFields = $this->Fields()->dataFields();
-        if (!$dataFields) {
+        // Don't include fields without data
+        $fields = $this->Fields();
+
+        if (!$fields || !$dataFields = $fields->dataFields()) {
```

Comment:
I guess the variable assignment isn't required any more...

Reply:
Mate, it was late. I'm cannng this one, and re-opening a new (the original) one. I'm conflating too many different issues in the one PR. Sorry.